src/NServiceBus.Transport.SQS/MessageDispatcher.cs

```diff
@@ -263,12 +263,9 @@ await sqsClient.SendMessageAsync(message.ToRequest(), cancellationToken)
             }
         }
 
-        async Task<TMessage> PrepareMessage<TMessage>(IOutgoingTransportOperation transportOperation, Dictionary<string, Type> multicastEventsMessageIdsToType, TransportTransaction transportTransaction, CancellationToken cancellationToken)
-            where TMessage : PreparedMessage, new()
+        async Task<SqsPreparedMessage> PrepareMessage(UnicastTransportOperation transportOperation, Dictionary<string, Type> multicastEventsMessageIdsToType, TransportTransaction transportTransaction, CancellationToken cancellationToken)
         {
-            var unicastTransportOperation = transportOperation as UnicastTransportOperation;
-
-            if (!await hybridPubSubChecker.PublishUsingMessageDrivenPubSub(unicastTransportOperation, multicastEventsMessageIdsToType, topicCache, queueCache, snsClient).ConfigureAwait(false))
+            if (!await hybridPubSubChecker.PublishUsingMessageDrivenPubSub(transportOperation, multicastEventsMessageIdsToType, topicCache, queueCache, snsClient).ConfigureAwait(false))
```

Comment:
I know that this isn't a new method but I found the name `PublishUsingMessageDrivenPubSub` confusing in this context. This PrepareMessage is preparing any message for sending, not just publishing messages using message driven pub sub, so what we are really checking is more like:

```
if (await hybridPubSubChecker.ThisIsAPublishMessageNotUsingMessageDrivenPubSub)
{
   return null;
}
```